### PR TITLE
feat: Added log package

### PR DESCRIPTION
**Pull request by patrickzheng200**
This PR adds log package to notation-go.

Signed-off-by: Patrick Zheng <patrickzheng@microsoft.com>

### CODE REVIEW

File: log/log.go [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+// Logger is implemented by users and/or 3rd party loggers.
+// For example, uber/zap.SugaredLogger and sirupsen/logrus.Logger.
+type Logger interface {
+	// Debug logs a debug level message.
+	Debug(args ...interface{})
+
+	// Debugf logs a debug level message with format.
+	Debugf(format string, args ...interface{})
+
+	// Debugln logs a debug level message with a newline appended.
```
[the remark this answers is not in the document]
Comment (patrickzheng200) [reply log/log.go:26]: > `xxxln` does not mean logs a message with a newline appended. For example, both `Info` and `Infoln` have no newline appended. In other words, `Info("foo")` and `Infoln("foo")` print exactly the same thing.
> 
> The difference is that `Info` invokes `fmt.Sprint` and `Infoln` invokes `fmt.Sprintln`.

From the func doc of [fmt.Sprintln](https://pkg.go.dev/fmt#Sprintln): Sprintln formats using the default formats for its operands and returns the resulting string. Spaces are always added between operands and a newline is appended.
A newline is appended, but Infoln trims this newline from the result of Sprintln. Will update the decritptions.

**Review by shizhMSFT — APPROVED** [review]
LGTM

File: log/log.go [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+type contextKey int
+
+// loggerKey is the associated key type for logger entry in context.
+const loggerKey contextKey = iota
+
+// Discard is a discardLogger that is used to disenable logging in notation.
+var Discard Logger = &discardLogger{}
+
+// Logger is implemented by users and/or 3rd party loggers.
+// For example, uber/zap.SugaredLogger and sirupsen/logrus.Logger.
```
Comment (shizhMSFT) [inline log/log.go:19]: nit: use full name `uber/zap.SugaredLogger` --> `github.com/uber/zap.SugaredLogger`.

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Review by rgnote — APPROVED** [review]
LGTM

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Pull request merged**